### DISCUSSION TITLE
eModbus to ESP-IDF migration

### DISCUSSION THREAD

**Discussion by brainstorm** [eModbus/eModbus#171] [eModbus applications]
Just in case somebody is interested in migrating from eModbus to ESP-IDF (in my case because ESP RainMaker is a WIP in Arduino), feel free to check out my journey and hiccups over here:

https://esp32.com/viewtopic.php?f=13&t=26953&p=94639#p94639

And here: https://github.com/brainstorm/yigedianqi-modbus/

Kudos to @Miq1 for helping me out a ton with the hardware side on https://github.com/eModbus/eModbus/discussions/166 ;)

**Comment by BrotherCreamy** [comment]
Any reason why one might want to use FreeModbus rather than eModbus generally?

Is FreeModbus more polished?

**Reply by brainstorm** [reply]
Not really, it's just that Espressif's cloud dev tools (ESP Insights) and their metrics hosting system (ESP Rainmaker) had some unfixed issues w/ Arduino (eModbus):

https://esp32.com/viewtopic.php?f=41&t=26552

I wanted to experiment with them and that's why I migrated. They have been pretty handy on provisioning and detecting memleaks and other bugs/reboots.

**Reply by YehorFedorov** [reply]
Let me join your discussion)

I have a case, when I need to implement the custom rts callback using freemodbus but eModbus provides an API for this - https://emodbus.github.io/modbusserver-rtu-api#modbusserverrtuhardwareserial-serial-uint32_t-timeout-rtscallback-func

@BrotherCreamy maybe you have an idea how to resolve this?
